vm-validator/src/vm_validator.rs

```diff
@@ -65,32 +91,26 @@ impl TransactionValidation for VMValidator {
         });
         use diem_vm::VMValidator;
 
-        let (version, state_root) = self.db_reader.get_latest_state_root()?;
-        let db_reader = Arc::clone(&self.db_reader);
-        let vm = self.vm.clone();
-
-        let smt = SparseMerkleTree::new(state_root);
-        let state_view = VerifiedStateView::new(
-            StateViewId::TransactionValidation {
-                base_version: version,
-            },
-            db_reader,
-            Some(version),
-            state_root,
-            &smt,
-        );
-
-        Ok(vm.validate_transaction(txn, &state_view))
+        Ok(self
+            .vm
+            .validate_transaction(txn, self.cached_state_view.borrow_state_view()))
     }
 
     fn restart(&mut self, config: OnChainConfigPayload) -> Result<()> {
         let vm_config = config.get::<VMConfig>()?;
         let version = config.get::<DiemVersion>()?;
         let publishing_option = config.get::<VMPublishingOption>()?;
+        let cached_state_view = CachedStateView::create(&self.db_reader);
+        self.cached_state_view = cached_state_view;
```

Comment:
Just put `self.notify_commit()` here.

Also, interesting, so I guess you're trying to not block the throughput, so your cached_state_view is replaced, rather than updated?